pkg/controller/common/metrics.go

```diff
@@ -81,7 +82,7 @@ func StartMetricsListener(addr string, stopCh <-chan struct{}, registerFunc func
 	klog.Infof("Starting metrics listener on %s", addr)
 	mux := http.NewServeMux()
 	mux.Handle("/metrics", promhttp.Handler())
-	s := http.Server{Addr: addr, Handler: mux}
+	s := http.Server{TLSNextProto: make(map[string]func(*http.Server, *tls.Conn, http.Handler)), Addr: addr, Handler: mux}
```

Comment:
Based on the conversation in slack, I think only the metrics server will be affected. Were you able to test whether this passes the listed tests for port 9001?